### PR TITLE
Set anim control and collision fixing

### PR DESCRIPTION
**Pull request by TeslaRus**
- added auto position correction for Entity_SetAnimation function: for hardcoded anim state controlling without hardcoded magic offsets;
 - fixed Entity_GetPenetrationFixVector algorithm;
 - enabled anti-tunneling feature for Entity_GetPenetrationFixVector;

### CODE REVIEW

File: src/anim_state_control.cpp [hunk 1]
```diff
@@ -2355,10 +2361,13 @@ int State_Control_Lara(struct entity_s *ent, struct ss_animation_s *ss_anim)
              * FREE FALL TO UNDERWATER CASES
              */
         case TR_STATE_LARA_UNDERWATER_DIVING:
-            ent->angles[1] = -45.0;
-            cmd->rot[1] = 0.0;
-            Entity_UpdateTransform(ent);
-            ss_anim->onEndFrame = ent_correct_diving_angle;
+            if(ent->move_type != MOVE_FREE_FALLING)
```
Comment (TeslaRus) [inline src/anim_state_control.cpp]: maybe that condition or whole section is unnecessary...

**Comment by vvs-** [comment]
Many entities has incorrect positions, e.g. doors and switches.

**Comment by TeslaRus** [comment]
wow! fast feedback! hope it https://github.com/opentomb/OpenTomb/commit/3b21f78997795578196383fcd60493d145de1856 helps

**Comment by vvs-** [comment]
So far so good. But the position at the end of ladder climbing is still incorrect. Also, there are problems with wading on slopes.

**Comment by TeslaRus** [comment]
Ladder fixing is another task (walls climbing only); that task is collision fixes fix and smooth positioning (hang to free fall: Lara falls down without up moving ~100 game units)
"Also, there are problems with wading on slopes." it is old bug meaning?

**Comment by vvs-** [comment]
> "Also, there are problems with wading on slopes." it is old bug meaning?

It's old, but this PR introduced new manifestations which I didn't see before.

Also, this PR breaks underwater propeller in TR2 _Offshore Rig_.

**Comment by TeslaRus** [comment]
I will test propellers... interesting... maybe reason of new bugs is that collision fixes works more correct so old bugs become visible (really before that PR was big mistake that prevents to "fix" out of room geometry).

**Comment by vvs-** [comment]
Sorry, it was my confusion. Actually, that propeller bug isn't new to this PR. This is happening because it's animated when flags are 0 instead of 31.

**Pull request merged**